drivers/flash/flash_npcx_fiu_nor.c

```diff
@@ -0,0 +1,550 @@
+	if (ret != 0) {
+		return ret;
+	}
+
+	return flash_npcx_nor_wait_until_ready(dev);
+}
+
+/* Flash API functions */
+#if defined(CONFIG_FLASH_JESD216_API)
+static int flash_npcx_nor_read_jedec_id(const struct device *dev, uint8_t *id)
+{
+	if (id == NULL) {
+		return -EINVAL;
+	}
+
+	return flash_npcx_uma_read(dev, SPI_NOR_CMD_RDID, id, SPI_NOR_MAX_ID_LEN);
+}
+
+static int flash_npcx_nor_read_sfdp(const struct device *dev, off_t addr,
+				    void *data, size_t size)
+{
+	uint8_t sfdp_addr[4];
+	struct npcx_uma_cfg cfg = { .opcode = JESD216_CMD_READ_SFDP,
+					.tx_buf = sfdp_addr,
+					.tx_count = 4,
+					.rx_buf = data,
+					.rx_count = size};
+
+	if (data == NULL) {
+		return -EINVAL;
+	}
+
+	/* CMD_READ_SFDP needs a 24-bit address followed by a dummy byte */
+	sfdp_addr[0] = (addr >> 16) & 0xff;
+	sfdp_addr[1] = (addr >> 8) & 0xff;
+	sfdp_addr[2] = addr & 0xff;
+	return flash_npcx_uma_transceive(dev, &cfg, NPCX_UMA_ACCESS_WRITE |
+					 NPCX_UMA_ACCESS_READ);
+}
+#endif /* CONFIG_FLASH_JESD216_API */
+
+#if defined(CONFIG_FLASH_PAGE_LAYOUT)
+static void flash_npcx_nor_pages_layout(const struct device *dev,
+					const struct flash_pages_layout **layout,
+					size_t *layout_size)
+{
+	const struct flash_npcx_nor_config *config = dev->config;
+
+	*layout = &config->layout;
+	*layout_size = 1;
+}
+#endif /* CONFIG_FLASH_PAGE_LAYOUT */
+
+static int flash_npcx_nor_read(const struct device *dev, off_t addr,
+				 void *data, size_t size)
+{
+	const struct flash_npcx_nor_config *config = dev->config;
+	uint32_t operation = ((struct flash_npcx_nor_data *)dev->data)->operation;
+
+	/* Out of the region of nor flash device? */
+	if ((addr < 0) || ((size + addr) > config->flash_size)) {
+		return -EINVAL;
+	}
+
+	/* Lock/Unlock SPI bus also for DRA mode */
+	qspi_npcx_fiu_mutex_lock_configure(config->qspi_bus, &config->qspi_cfg, operation);
+
+	/* Trigger Direct Read Access (DRA) via reading memory mapped-address */
+	memcpy(data, (void *)(config->mapped_addr + addr), size);
+
+	qspi_npcx_fiu_mutex_unlock(config->qspi_bus);
+
+	return 0;
+}
+
+static int flash_npcx_nor_erase(const struct device *dev, off_t addr,
+				  size_t size)
+{
+	const struct flash_npcx_nor_config *config = dev->config;
+	int ret = 0;
+
+	/* Out of the region of nor flash device? */
+	if ((addr < 0) || ((size + addr) > config->flash_size)) {
+		return -EINVAL;
+	}
+
+	/* address must be sector-aligned */
+	if (!SPI_NOR_IS_SECTOR_ALIGNED(addr)) {
+		return -EINVAL;
+	}
+
+	/* size must be a multiple of sectors */
+	if ((size % config->min_erase_size) != 0) {
+		return -EINVAL;
+	}
+
+	while (size > 0) {
+		flash_npcx_uma_cmd_only(dev, SPI_NOR_CMD_WREN);
+
+		if (size == config->flash_size) {
+			/* chip erase */
+			flash_npcx_uma_cmd_only(dev, SPI_NOR_CMD_CE);
+			size -= config->flash_size;
+		} else {
+			uint8_t opcode;
+
+			if (config->min_erase_size == 0x1000) {
+				opcode = SPI_NOR_CMD_SE;
+			} else if (config->min_erase_size == 0x10000) {
```

Comment:
How about a pair of `#define` for these?

Reply:
Done.